CIP-0062/README.md

```diff
@@ -0,0 +1,429 @@
+
+This also addresses some short-comings of [CIP-30](https://cips.cardano.org/cips/cip30/); which signData can only be done by known an address; This signature is not relevant to a specific address, nor the dApp will know an address attached to the voting key. The voting key derivation is defined in [CIP-36](https://cips.cardano.org/cips/cip36/). 
+
+Perhaps [CIP-30](https://cips.cardano.org/cips/cip30/) could be expanded to also know how to perform `signData` from a given public key from a specific derivation path; instead of doing so only by known address.
+
+The other reason for this specification is to have a specific, but optional, namespace for governance specific wallet functionality. As such, wallet providers might choose not to implement this specification on top of [CIP-30](https://cips.cardano.org/cips/cip30/).
+
+# Specification
+
+## Version
+
+The API Extension specified in this document will count as version 0.2.0 for version-checking purposes below.
+
+## Data Types
+
+### PublicKey
+
+TODO: Define this.
+
+### GovernanceKey
+
+```ts
+type GovernanceKey = {
+  votingKey: string
+  weight: number
+}
+```
+
+* `votingKey` - Ed25519 pubkey 32 bytes HEX string
+* `weight` - Used to calculate the actual voting power using the rules described
+  in [CIP-36](https://cips.cardano.org/cips/cip36/).
+
+### Voting Purpose
+
+```ts
+type enum VotingPurpose = {
+  CATALYST = 0
+}
+```
+
+`VotingPurpose`: Defines the voting purpose of the governance functions. This is used
+to limit the scope of the governance functions. For example, a voting purpose
+might be a subset of Catalyst proposals, a council election, or even some
+private purpose (agreed by convention). Currently, only voting purpose 0 (Zero)
+is defined, and it is for Catalyst events.
+
+**IMPORTANT**: The List of purposes in this CIP should be considered the
+authoritative list of known purposes, subject to future amendment. Other voting
+purposes will be defined as required, by either an update to this CIP or a
+future CIP listing currently allocated Voting Purposes.
+
+### BlockDate
+
+```ts
+interface BlockDate {
+  epoch: number
+  slot: number
+}
+```
+
+* epoch - an epoch value on the voting block chain.
+* slot - a block slot number on the voting block chain.
+
+#### Proposal
+
+```ts
+interface Proposal {
+  votePlanId: string
+  voteOptions: number[]
+  votePublic: boolean
+  proposalIndex: number
+}
+```
+
+Proposal information.
+
+* votePlanId - Hex encoded string to represent vote plan unique identifier. This will be the same as anchored on voting chain, and is managed by the dApp.
+* voteOptions - List of possible "choices" which can be made for this proposal.
+* votePublic - Boolean indicating whether the vote is public or private.
+
+### Vote
+
+```ts
+interface Vote {
+  proposal: Proposal
+  choice: number
+  expiration: BlockDate
+  purpose: VotingPurpose
+  spendingCounter: number
```

Comment:
I think also wallet settings are missing here. The vote signer needs to know some basic network parameters to produce a valid vote. It is a stringified JSON in the vote signing lib: 

https://github.com/input-output-hk/catalyst-core/blob/main/src/chain-wallet-libs/bindings/wallet-js/js-test/test/vote_cast.js#L9

Reply:
also, I'm not sure whether some parameter to manage the "lanes" wouldn't be needed in case of signing multiple votes with the `signVotes()` call: https://github.com/input-output-hk/catalyst-core/issues/131#issuecomment-1339195034